zingolib/src/wallet/traits.rs

```diff
@@ -1423,9 +1389,22 @@ where
         writer.write_all(&self.diversifier().to_bytes())?;
 
         self.note().write(&mut writer)?;
-        writer.write_u64::<LittleEndian>(u64::from(*self.witnessed_position()))?;
-
-        writer.write_all(&self.nullifier().to_bytes())?;
+        writer.write_u64::<LittleEndian>(u64::from(self.witnessed_position().ok_or(
+            io::Error::new(
+                io::ErrorKind::InvalidData,
+                "Tried to write note with unknown position",
+            ),
+        )?))?;
+
+        writer.write_all(
+            &self
+                .nullifier()
+                .ok_or(io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    "Tried to write note with unknown nullifier",
```

Comment:
I agree with @AArnott that it's high time to standardize an Error Enum that covers all the `zingolib` cases...

Reply:
truh, but thats for another PR